File: scripts.js
document.addEventListener('DOMContentLoaded', function() {
    const darkModeToggle = document.getElementById('dark-mode-toggle');
    const body = document.body;
    let slideIndex = 0;
    const slides = document.getElementsByClassName("mySlides");
    const dots = document.getElementsByClassName("dot");

    function toggleDarkMode() {
        body.classList.toggle('dark-mode');
        if (body.classList.contains('dark-mode')) {
            darkModeToggle.innerHTML = '<i class="fas fa-sun"></i> Light Mode';
            document.querySelectorAll('.social-links a img').forEach(img => {
                img.classList.add('invert-icon');
            });
        } else {
            darkModeToggle.innerHTML = '<i class="fas fa-moon"></i> Dark Mode';
            document.querySelectorAll('.social-links a img').forEach(img => {
                img.classList.remove('invert-icon');
            });
        }
    }

    if (darkModeToggle) {
        darkModeToggle.addEventListener('click', toggleDarkMode);
    } else {
        console.error('Dark mode toggle button element not found.');
    }

    body.classList.remove('dark-mode');
    darkModeToggle.innerHTML = '<i class="fas fa-moon"></i> Dark Mode';
    document.querySelectorAll('.social-links a img').forEach(img => {
        img.classList.remove('invert-icon');
    });

    function showSlides(n) {
        if (n >= slides.length) { slideIndex = 0; }
        if (n < 0) { slideIndex = slides.length - 1; }
        for (let i = 0; i < slides.length; i++) {
            slides[i].style.display = "none";
        }
        slides[slideIndex].style.display = "block";

        for (let i = 0; i < dots.length; i++) {
            dots[i].classList.remove("active");
        }
        dots[slideIndex].classList.add("active");
    }

    function nextSlide() {
        slideIndex++;
        showSlides(slideIndex);
    }

    function previousSlide() {
        slideIndex--;
        showSlides(slideIndex);
    }

    function autoSlides() {
        slideIndex++;
        if (slideIndex >= slides.length) { slideIndex = 0; }
        showSlides(slideIndex);
        setTimeout(autoSlides, 2000); 
    }

    autoSlides();

    document.getElementById('prev').addEventListener('click', function() {
        previousSlide();
    });
    document.getElementById('next').addEventListener('click', function() {
        nextSlide();
    });

    for (let i = 0; i < dots.length; i++) {
        dots[i].addEventListener("click", function() {
            slideIndex = i;
            showSlides(slideIndex);
        });
    }
});

File: scripts.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>
#include <unistd.h> 


typedef struct {
    char *className;
    bool display;
    bool active;
} Slide;

typedef struct {
    bool invert_icon;
} Img;

typedef struct {
    Slide *slides;
    int slides_count;
    Img *imgs;
    int imgs_count;
    int slideIndex;
    bool dark_mode;
} Document;

void toggleDarkMode(Document *doc) {
    doc->dark_mode = !doc->dark_mode;
    if (doc->dark_mode) {
        printf("Dark mode enabled: showing sun icon and Light Mode text\n");
        for (int i = 0; i < doc->imgs_count; i++) {
            doc->imgs[i].invert_icon = true;
        }
    } else {
        printf("Dark mode disabled: showing moon icon and Dark Mode text\n");
        for (int i = 0; i < doc->imgs_count; i++) {
            doc->imgs[i].invert_icon = false;
        }
    }
}

void showSlides(Document *doc, int n) {
    if (n >= doc->slides_count) {
        doc->slideIndex = 0;
    } else if (n < 0) {
        doc->slideIndex = doc->slides_count - 1;
    } else {
        doc->slideIndex = n;
    }

    for (int i = 0; i < doc->slides_count; i++) {
        doc->slides[i].display = false;
        doc->slides[i].active = false;
    }
    doc->slides[doc->slideIndex].display = true;
    doc->slides[doc->slideIndex].active = true;

    printf("Showing slide %d\n", doc->slideIndex);
}

void nextSlide(Document *doc) {
    showSlides(doc, doc->slideIndex + 1);
}

void previousSlide(Document *doc) {
    showSlides(doc, doc->slideIndex - 1);
}

void autoSlides(Document *doc) {
    while (true) {
        nextSlide(doc);
        sleep(2);
    }
}

int main() {
   
    int slides_count = 3;
    Slide slides[3] = {
        {.className = "mySlides", .display = false, .active = false},
        {.className = "mySlides", .display = false, .active = false},
        {.className = "mySlides", .display = false, .active = false}
    };

    int imgs_count = 2;
    Img imgs[2] = {
        {.invert_icon = false},
        {.invert_icon = false}
    };

    Document doc = {
        .slides = slides,
        .slides_count = slides_count,
        .imgs = imgs,
        .imgs_count = imgs_count,
        .slideIndex = 0,
        .dark_mode = false
    };

    doc.dark_mode = false;
    for (int i = 0; i < imgs_count; i++) {
        doc.imgs[i].invert_icon = false;
    }
    printf("Initial mode: Dark Mode off, showing moon icon\n");

    toggleDarkMode(&doc);
    toggleDarkMode(&doc);

    showSlides(&doc, 0);
    nextSlide(&doc);
    previousSlide(&doc);

    return 0;
}